src/main.controller.js

```diff
@@ -440,6 +446,22 @@ plugin.controller('wgnConfigCtrl', ['$scope', '$q', '$routeParams', 'znData', 'z
 			}
 		};
 
+		/**
+		 * Reset choice values when source field changes
+		 *
+		 * @param {fieldId} fieldDef The choice input id
+		 *
+		 */
+		$scope.resetChoiceValues = function(fieldId) {
```

Comment:
@AParks have you verified 100% that deleting a value and saving it actually removes it from firebase?  for some reason i want to say that it might be better to set the value to `null` and that would have the same effect in the end...

Reply:
yes I'm 100% sure, I've tested it. I think you might be thinking of the [$firebase.$update method](https://zenginehq.github.io/developers/libraries/angularfire/0.8.0/#angularfire-firebase-updatekey-data):

> Any child keys specified in data are completely replaced by the values provided (update only works one level deep into child records, not recursively), and any child keys not in data are left alone.

But in the frontend, we normally work with [$FirebaseObject.$save](https://zenginehq.github.io/developers/libraries/angularfire/0.8.0/#angularfire-firebaseobject-save). It's more likely you came across this update behavior in the backend.